(dp1
S'output'
p2
S"<type 'exceptions.AttributeError'> class Validator has no attribute 'process_form'"
p3
sS'layer'
p4
S'/home/filhofilha/Documents/Deployment/Web2Py/web2py/applications/perimoveis/controllers/product.py'
p5
sS'code'
p6
S"# -*- coding: utf-8 -*-\n\n\n\n\ndef index():\n\n    return dict(mRows=db(db.tbProducts.id>0).select())\n\n\n\n@auth.requires_login()\ndef add():\n    Validator.admin()\n\n    mForm=SQLFORM(db.tbProducts, submit_button=T('Add a product'))\n\n    Validator.process_form(mForm, URL('product', 'index'))\n\n    return dict(mForm=mForm)\n\n\n\n@auth.requires_login()\ndef edit():\n    Validator.admin()\n    Validator.valide_args(1)\n\n    return dict(mForm=SQLFORM(db.tbProducts,\n                db.tbProducts(request.args[0]),\n                submit_button=T('Save changes')))\n\nresponse._vars=response._caller(add)\n"
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/filhofilha/Documents/Deployment/Web2Py/web2py/gluon/restricted.py", line 227, in restricted\n    exec ccode in environment\n  File "/home/filhofilha/Documents/Deployment/Web2Py/web2py/applications/perimoveis/controllers/product.py", line 33, in <module>\n  File "/home/filhofilha/Documents/Deployment/Web2Py/web2py/gluon/globals.py", line 412, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/filhofilha/Documents/Deployment/Web2Py/web2py/gluon/tools.py", line 4236, in f\n    return action(*a, **b)\n  File "/home/filhofilha/Documents/Deployment/Web2Py/web2py/applications/perimoveis/controllers/product.py", line 18, in add\n    Validator.process_form(mForm, URL(\'product\', \'index\'))\nAttributeError: class Validator has no attribute \'process_form\'\n'
p11
s.